woocommerce/woocommerce
# Add order attribution install banner in order editor's order attribution metabox

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Part of #51567, this PR adds the order attribution install banner in the order editor's order attribution metabox. In contrast to the main banner (#51672) and the header banner (#51908) on the analytics overview page, which are displayed depending on whether the main banner is dismissed, this banner will appear regardless of the main banner's dismissal status.

Note that when clicking **Install the extension** on the banner, it will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found as the plugin hasn't been published.

<img width="1264" alt="Screenshot 2024-10-24 at 11 15 33" src="https://github.com/user-attachments/assets/bb029364-efec-481e-a6c5-f91025346be7">

#### Technical details

- I refer to "Launch your store settings" to inject the React component to the order attribution metabox (PHP), by using the SlotFill mechanism from WordPress components. Ref: [1](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/settings/settings-slots.js) and [2](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/launch-your-store/settings/slotfill.js).
- This PR also moves the code from `plugins/woocommerce-admin/client/order-attribution-install-banner` to `plugins/woocommerce/client/admin/client/order-attribution-install-banner` in favour of #49647.

#### Event tracking

Same as in #51672 and #51908, this PR sends two events:
- `order_attribution_install_banner_viewed`
- `order_attribution_install_banner_clicked`

But with different `context` property: `order-editor-order-attribution-metabox`.

#### Next steps

- Display the banner in other places within WooCommerce Admin
- Implement displaying the banner for % of merchants
- The changelog entry will be added in the final PR once all the banners are added



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the Console tab in the browser DevTool, and run `localStorage.setItem( 'debug', 'wc-admin:*' )` to enable tracking debugging mode
2. Do not install `woocommerce-analytics` plugin yet
3. Go to any order, in editing mode
4. You should see the install banner appears under the order attribution metabox in the right-hand side
5. The event `wcadmin_order_attribution_install_banner_viewed` with `context: order-editor-order-attribution-metabox` should be sent
6. Go to `Analytics > Overview`, dismiss the main banner
7. Go back to the order editor, you should still see the install banner appears under the order attribution metabox in the right-hand side
8. Click **Install the extension**, it will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found
9. The event `wcadmin_order_attribution_install_banner_clicked` with `context: order-editor-order-attribution-metabox` should be sent
10. Install `woocommerce-analytics` plugin, do not activate it
11. Go to the order editor, the banner should not be shown
12. Activate `woocommerce-analytics` plugin
13. Go to the order editor, the banner should not be shown
14. ~Go to `Analytics > Overview`, ensure the main banner and the header banner should still work as expected~ Remove `woocommerce-analytics` plugin and go back to `Analytics > Overview`, ensure the main banner and the header banner should still work as expected after this PR is made.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
